library/core/src/iter/adapters/intersperse.rs

```diff
@@ -0,0 +1,76 @@
+use super::Peekable;
+
+/// An iterator adapter that places a separator between all elements.
```

Comment:
```suggestion
/// An iterator adapter that places a separator between all elements.
///
/// This `struct` is created by [`Iterator::intersperse`]. See its
/// documentation for more.
```
Document how was in created.